libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/building/BuildingProcessor.kt

```diff
@@ -14,12 +14,16 @@ internal object BuildingProcessor {
      * Source layer that includes building height.
      */
     private const val BUILDING_LAYER_ID = "building"
+    private const val BUILDING_EXTRUSION_LAYER_ID = "building-extrusion"
```

Comment:
🤔  if we can use regular expression. When someone creates style, in my understanding they can call this layer something else other than `building` or `building-extrusion` . In that case this solution would still not work and we would probably have to add another string to that list.
I have asked this question [here](https://mapbox.slack.com/archives/C02KT0MGUSC/p1647369643110209)

Reply:
The terrain layer uses `building-extrusion` so the values we're using cover Mapbox styles.  Ideally there would be an option that developers could use to indicate the layer(s) to search.

Reply:
The issue was asking for parity between Android and iOS, which we're not precisely there yet. [Here's the relevant iOS code ](https://github.com/mapbox/mapbox-navigation-ios/blob/3669b03e776b5910ac6976965deb5f01b4a86418/Sources/MapboxNavigation/NavigationMapView%2BBuildingHighlighting.swift#L28-L30) which seems to be looking for a `"building"` substring in a layer name, why don't we do the same?
cc @MaximAlien in case there's something else we need to consider

Reply:
Check for layers that contain `building` is more generic, as we'll pick the one that is the last one and add our layer above it. At the same time querying for data that is contained in the layer will be done for all layers that contain `building` in their identifiers. 

Not sure that it's worth exposing the ability to provide a way for the user to set their own identifiers where they want to perform search (too much manual and not obvious work).